### DISCUSSION TITLE
Trying to understand gain, peak and clipping

### DISCUSSION THREAD

**Discussion by NaanProphet** [complexlogic/rsgain#103] [Q&A] [answered]
This is kind of a ReplayGain fundamental question.

I'm not sure how to quite interpret the results below. If I were to apply the Gain suggested, the track would be `-25.93 LUFS + 7.93 dB = -18 LUFS`. However, the peak is practically already 1.0, which means applying the gain would make the track clip right?

```
% ./rsgain custom --album *
[✔] Scanning 'Track 01.m4a'
[✔] Container: QuickTime / MOV [mov,mp4,m4a,3gp,3g2,mj2]
[✔] Stream #0: ALAC (Apple Lossless Audio Codec), 16 bit, 44,100 Hz, 2 ch
 100% [=========================================================================]

Track: Track 01.m4a
  Loudness:   -25.93 LUFS
  Peak:     0.999390 (-0.01 dB)
  Gain:         7.93 dB 
```

**Comment by complexlogic** [comment, marked as answer]
Generally, yes, that is the case. If the peak of the audio signal is at or close to digital full scale, and there is a large positive gain calculated, that would result in clipping if the gain is applied during playback.

It may not actually clip if the player is not set at the max volume, though, or if there is other attenuation of the signal during playback. That's why it's best if the clipping protection is implemented in the player, because it has information that is impossible to know at scan time.
